node/network/protocol/src/peer_set.rs

```diff
@@ -102,50 +111,35 @@ impl PeerSet {
 		}
 	}
 
-	/// Get the default protocol version for this peer set.
-	pub const fn get_default_version(self) -> ProtocolVersion {
-		match self {
-			PeerSet::Validation => DEFAULT_VALIDATION_PROTOCOL_VERSION,
-			PeerSet::Collation => DEFAULT_COLLATION_PROTOCOL_VERSION,
-		}
+	/// Get the main protocol version for this peer set.
+	///
+	/// Networking layer relies on `get_main_version()` being the version
+	/// of the main protocol name reported by [`PeerSetProtocolNames::get_main_name()`].
+	pub const fn get_main_version(self) -> ProtocolVersion {
+		MAIN_PROTOCOL_VERSION
 	}
 
-	/// Get the default protocol name as a static str.
-	pub const fn get_default_protocol_name(self) -> &'static str {
-		match self {
-			PeerSet::Validation => VALIDATION_PROTOCOL_V1,
-			PeerSet::Collation => COLLATION_PROTOCOL_V1,
-		}
+	/// Get the max notification size for this peer set.
+	pub fn get_max_notification_size(self, _: IsAuthority) -> u64 {
+		MAX_NOTIFICATION_SIZE
 	}
 
-	/// Get the protocol name associated with each peer set
-	/// and the given version, if any, as static str.
-	pub const fn get_protocol_name_static(self, version: ProtocolVersion) -> Option<&'static str> {
-		match (self, version) {
-			(PeerSet::Validation, 1) => Some(VALIDATION_PROTOCOL_V1),
-			(PeerSet::Collation, 1) => Some(COLLATION_PROTOCOL_V1),
-			_ => None,
+	/// Get the peer set label for metrics reporting.
+	pub fn get_label(self) -> &'static str {
+		match self {
+			PeerSet::Validation => "validation",
+			PeerSet::Collation => "collation",
```

Comment:
May be it is worth keeping this `"/polkadot/validation/1"` & `"/polkadot/collation/1"` for backward compatibility? And also in `get_protocol_label()` below.

Reply:
I think it is fine. You are probably not really interested in prometheus if someone is still on the old version.